awsconfigs/apps/kserve/inferenceservice.yaml

```diff
@@ -0,0 +1,13 @@
+apiVersion: v1
+kind: ConfigMap
+metadata:
+  name: inferenceservice-config
+  namespace: kubeflow
+data:
+  credentials: |-
+    {
+       "s3": {
```

Comment:
Have you tried using [IRSA](https://docs.aws.amazon.com/eks/latest/userguide/iam-roles-for-service-accounts.html). I believe if you're using in production its better to use IAM role than static credential. 

There has been some discussion around [here](https://docs.aws.amazon.com/eks/latest/userguide/iam-roles-for-service-accounts.html) however I am not sure if the feature works in KServe. It would be worth testing it.

Reply:
@goswamig I have not tried using IRSA with KFServing/KServe, I know KFServing/KServe expects the AWS environment variables (`AWS_ACCESS_KEY_ID`, `AWS_SECRET_ACCESS_KEY`) to be available for the `storage-initializer`, not sure if IRSA works or how to set it up. I agree that long term IRSA is the best approach, but it doesn't seem to be fully documented yet.  

I went for the minimal approach here since currently [KServe documentation](https://kserve.github.io/website/modelserving/storage/s3/s3/) only showcase one way (via secret + service account creation)

Reply:
Right, what I am saying is instead of creating secrets and passing its name to service account. Can we rather just annotate the service account with a role which has policy attached for S3 download/upload? This will help us to at least rule out the feature is broken.

Reply:
@goswamig I'm open to it for sure, I'd just need more guidance.  
  
Would it be [something like this ](https://aws.amazon.com/premiumsupport/knowledge-center/eks-restrict-s3-bucket/?nc1=h_ls)?

Reply:
Given that user (under profiler) will create `InferenceService`, you can annotate your profile with IAM service which will have attached policy to READ/Write S3. 

For e.g. https://github.com/kubeflow/kubeflow/blob/master/components/profile-controller/config/samples/profile_v1_aws_iam.yaml#L9